icu4j/main/core/src/main/java/com/ibm/icu/message2/Function.java

```diff
@@ -41,4 +42,24 @@ public interface Formatter {
      */
     @Deprecated
     FormattedPlaceholder format(Object toFormat, Map<String, Object> variableOptions);
+
+    /**
+     * A method that is invoked for the object to match and each key.
+     *
+     * <p>For example an English plural {@code matches} would return {@code true}
+     * for {@code matches(1, "1")}, {@code matches(1, "one")}, and {@code matches(1, "*")}.</p>
+     *
+     * @param value the value to select on.
+     * @param keys the key to test for matching.
+     * @param variableOptions options that are not know at build time.
+     * @return the formatted string.
+     *
+     * @internal ICU 72 technology preview
+     * @deprecated This API is for technology preview only.
+     */
+    @Deprecated
+    default List<String> matches(Object value, List<String> keys, Map<String, Object> variableOptions) {
+        // TODO: TZU
```

Comment:
not sure what TZU means. did you want to keep this line? is there a ticket number or description of what you wanted to include here, or is TZU enough?